python/ray/tune/suggest/suggestion.py

```diff
@@ -15,6 +15,18 @@
     "conversion, pass the space definition as part of the `config` argument "
     "to `tune.run()` instead.")
 
+UNDEFINED_SEARCH_SPACE = str(
+    "Trying to sample a configuration from {cls}, but no search "
+    "space has been defined. Either pass the `{space}` argument when "
+    "instantiating the search algorithm, or pass a `config` to "
+    "`tune.run()`.")
+
+UNDEFINED_METRIC_MODE = str(
```

Comment:
I think if we want to be really clear I would say splitting this up into two errors- one for metric and one for mode might make sense.

Reply:
My concern is that if someone does not pass a metric, he probably does not pass a mode as well (since they kind of go hand in hand). Thus I'd like to display both values in any case to make sure that the user fixes the error right away and provides both values. I'd think this is easier with just one message, or do you think it's better to split this up?

Reply:
Got it ok, that makes sense.